### PR TITLE
API controller: add API validation - allow one API per service

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- api-controller validates if an API already exists for a specified Kubernetes service
- tests

### CODE REVIEW

File: components/api-controller/pkg/controller/v1alpha2/controller.go [hunk 1]
```diff
@@ -204,22 +206,59 @@ func (c *Controller) onCreate(api *kymaApi.Api) error {
...
+	}
+
+	targetServiceName := newAPI.Spec.Service.Name
+
+	existingAPIs, err := c.apisLister.List(labels.Everything())
+	if err != nil {
+		log.Errorf("Error while listing APIs %s/%s ver: %s. Root cause: %s", newAPI.Namespace, newAPI.Name, newAPI.ResourceVersion, err)
+		return setStatus(kymaMeta.Error)
+	}
+
```
Comment (piotrmsc) [inline components/api-controller/pkg/controller/v1alpha2/controller.go]: You need to validated api for service in the given namespace of the created api

File: components/api-controller/pkg/apis/gateway.kyma-project.io/v1alpha2/types.go [hunk 2]
```diff
@@ -46,31 +46,37 @@ type JwtAuthentication struct {
...
 	AuthenticationStatus kymaMeta.GatewayResourceStatus `json:"authenticationStatus,omitempty"`
 	VirtualServiceStatus kymaMeta.GatewayResourceStatus `json:"virtualServiceStatus,omitempty"`
 }
 
 func (s *ApiStatus) IsEmpty() bool {
-	return s.VirtualServiceStatus.IsEmpty() && s.AuthenticationStatus.IsEmpty()
+	return s.VirtualServiceStatus.IsEmpty() && s.AuthenticationStatus.IsEmpty() && s.ValidationStatus.IsEmpty()
 }
 
 func (s *ApiStatus) IsDone() bool {
```
Comment (piotrmsc) [inline components/api-controller/pkg/apis/gateway.kyma-project.io/v1alpha2/types.go]: according to renaming should be `isSuccessful` here as well :)

**Comment by piotrmsc** [comment]
If you create an API for a service that does not have API attached but you have selected wrong service and make an update to correct one, Virtual service and Policy(if set) should be deleted for the wrong one (the first selected)

**Comment by piotrmsc** [comment]
Please extend our api-controller-acceptnce-tests with assertions for validationStatus.
Please also create an integration test for scenarios
And while chaning that please consider making this line https://github.com/kyma-project/kyma/blob/master/tests/api-controller-acceptance-tests/apicontroller/component_test.go#L252
configurable via env like it is with domain for example

**Comment by jakkab** [comment]
@piotrmsc Virtual Services and Polices are updated as well, no need to do delete anything.

Follow-up will be created.

**Pull request merged**